modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -297,6 +309,13 @@ private void undeploy(T artifact, boolean removeData) {
       deploymentListener.onUndeploymentStart(artifact.getArtifactName());
 
       artifacts.remove(artifact);
+      Optional<Registry> optionalRegistry = ofNullable(artifact.getRegistry());
+
+      optionalRegistry.ifPresent(artifactRegistry -> {
+        Optional<ArtifactStoppedListener> optionalArtifactStoppedListener =
+            artifactRegistry.lookupByName(ARTIFACT_STOPPED_LISTENER);
+        optionalArtifactStoppedListener.ifPresent(artifactStoppedListener -> artifactStoppedListener.mustPersist(false));
```

Comment:
no need to use the intermediate Optional just for this. a simple null check will do.